sdk/storage/azure-storage-datamovement/src/job_engine_task_generation.cpp

```diff
@@ -127,7 +128,63 @@ namespace Azure { namespace Storage { namespace _detail {
     }
     else if (m_model.Source.m_type == _internal::TransferEnd::EndType::AzureBlobFolder)
     {
-      AZURE_NOT_IMPLEMENTED();
+      const std::string rootDirectory = _internal::PathFromUrl(m_model.Destination.m_url);
+      std::string currentDirectory = rootDirectory;
+
+      partGens.push_back(gen);
+      if (!_internal::IsDirectory(currentDirectory))
+      {
+        _internal::CreateDirectory(currentDirectory);
+        // TODO: error handling
+      }
+
+      do
+      {
+        PartGeneratorModel currGen = std::move(partGens.back());
+        partGens.pop_back();
+
+        Blobs::ListBlobsOptions options;
+        options.Prefix = m_model.Source.m_blobFolder.Value().GetFolderPath();
+        options.PageSizeHint = 250;
+        options.ContinuationToken = std::move(currGen.ContinuationToken);
+
+        std::vector<_internal::Task> subtasks;
+        auto result = m_model.Source.m_blobFolder.Value().GetContainerClient().ListBlobs(options);
+        for (auto blobItem : result.Blobs)
+        {
+          TaskModel task;
+          const std::string blobName = blobItem.Name.substr(options.Prefix->length());
+          std::string localFileName = blobName;
+
+          std::string parentDir
+              = _internal::JoinPath(rootDirectory, _internal::GetParentDir(localFileName));
+          if (parentDir.compare(currentDirectory) != 0)
+          {
+            if (!_internal::IsDirectory(parentDir))
+            {
+              _internal::CreateDirectory(parentDir);
```

Comment:
Does `_internal::CreateDirectory` create directories recursively? I mean if `rootDirectory` is `/root` and `parentDir` is `/root/a/b/c/d`, does this still work?

Reply:
Added recursive create function